### DISCUSSION TITLE
[Unity]执行Generate Code/打包时静态wrap代码报错'XXXX' does not contain a definition fro 'xxxxx'

### DISCUSSION THREAD

**Discussion by throw-out** [Tencent/puerts#806] [Show and tell]
# 报错截图:
![~HSD9W_6U7IM15I)F 16D)V](https://user-images.githubusercontent.com/45587825/167990338-1caf7a9d-279a-4a5d-8182-ff30f50a7e0e.jpg)

# 解决方案
配置[Filter标签](https://github.com/Tencent/puerts/blob/master/unity/Assets/core/upm/Editor/Src/Configure.cs#L56)来告诉puerts不要生成这个Member

> 文档说明: [FAQ说明](https://github.com/Tencent/puerts/blob/master/doc/unity/zhcn/faq.md#%E7%BC%96%E8%BE%91%E5%99%A8%E4%B8%8B%E8%BF%90%E8%A1%8C%E6%AD%A3%E5%B8%B8%E6%89%93%E5%8C%85%E7%9A%84%E6%97%B6%E5%80%99%E7%94%9F%E6%88%90%E4%BB%A3%E7%A0%81%E6%8A%A5%E6%B2%A1%E6%9C%89%E6%9F%90%E6%96%B9%E6%B3%95%E5%B1%9E%E6%80%A7%E5%AD%97%E6%AE%B5%E5%AE%9A%E4%B9%89%E6%80%8E%E4%B9%88%E5%8A%9E)
> 示例: [demo示例](https://github.com/chexiongsheng/puerts_unity_demo/blob/master/projects/1_Start_Template/Assets/Samples/Editor/01_WrapperGenerate/ExamplesCfg.cs#L74) | [demo黑名单示例](https://github.com/chexiongsheng/puerts_unity_demo/blob/master/projects/1_Start_Template/Assets/Samples/Editor/01_WrapperGenerate/PuertsFilter.cs#L21)

# 为什么会出现报错
- Generate Code使用的是Unity Editor程序集, 而生成的静态wrap代码属于Runtime程序集。
因Unity版本差异和Api Compatibility Level设置不一样, Unity Editor与Runtime程序集相比在APi会有细微的差别(通常Unity Editor有额外的Member)
- 打包时报错则是Member代码被宏编译包含, 目标打包平台不满足该宏编译条件(例如: UNITY_EDITOR)

**Comment by zombieyang** [comment]
@throw-out 我在想，如果可以进入PlayMode后生成代码，是不是就没那么多问题了。在2019+好像是有这样的API的，unity-test-framework就有这么干。

**Reply by throw-out** [reply]
@zombieyang PlayMode不会影响Unity Editor程序集, 刚刚试过了
